aten/src/ATen/native/native_functions.yaml

```diff
@@ -611,13 +611,10 @@
     CPU: _sum_cpu
     CUDA: _sum_cuda
 
-- func: sum(Tensor self, int64_t dim, bool keepdim=False) -> Tensor
+- func: sum(Tensor self, IntList[1] dim, bool keepdim=False) -> Tensor
```

Comment:
What does `IntList[1]` mean?

Reply:
`Ìntlist[1]` means that if you pass an Integer, it will be converted to a 1-vector containing that integer (prominently used for convs).

The issue here is (briefly touched in the initial comments) that the JIT does not allow overloading of the form that one function takes an int and another an IntList. Thus, I used this to reroute the one-dimensional sum.

The alternative would be to change `gen_jit_dispatch.py` to consider the two signatures to be distinct. Currently I get exceptions in emit_decl_variant - the assertion at the bottom of that function if I have both `func: sum(Tensor self, int64_t dim, bool keepdim=False) -> Tensor` and
`func: sum(Tensor self, IntList[1] dim, bool keepdim=False) -> Tensor`.

Reply:
That’s actually surprising, because those should be two different signatures (one uses an `i` attr, while the other uses `is`)

Reply:
But this works too (and I think I prefer it)!